### PR TITLE
Add configuration round-trip testing

### PR DESCRIPTION
**Pull request by sectioneight**
This verifies that our descriptors are properly configured and that the
getters/setters round-trip through the bean.

https://wiki.jenkins-ci.org/display/JENKINS/Unit+Test#UnitTest-Configurationroundtriptesting

Had to add a hamcrest dependency to get the newer (1.3) matchers.

### CODE REVIEW

**Comment by sectioneight** [comment]
@jjx -- not sure if there's a good way to DRY up the tests since they're different kinds -- I tend to prefer a little bit of duplication in tests for clarity, lmk if you have any ideas.

File: phabricator-plugin.iml [hunk 1]
```diff
@@ -124,8 +124,9 @@
     <orderEntry type="library" name="Maven: net.java.dev.jna:jna:3.2.2" level="project" />
     <orderEntry type="library" scope="TEST" name="Maven: org.mockito:mockito-core:2.0.23-beta" level="project" />
     <orderEntry type="library" scope="TEST" name="Maven: net.bytebuddy:byte-buddy:0.6.11" level="project" />
-    <orderEntry type="library" scope="PROVIDED" name="Maven: org.hamcrest:hamcrest-core:1.1" level="project" />
     <orderEntry type="library" scope="TEST" name="Maven: org.objenesis:objenesis:2.1" level="project" />
+    <orderEntry type="library" scope="TEST" name="Maven: org.hamcrest:hamcrest-core:1.3" level="project" />
+    <orderEntry type="library" scope="TEST" name="Maven: org.hamcrest:hamcrest-library:1.3" level="project" />
```
Comment (sectioneight) [inline phabricator-plugin.iml]: is this file supposed to be gitignored?

**Comment by jjx** [comment]
Totally cool with the duplication. Btw, why do we need the new dependencies?

**Comment by sectioneight** [comment]
Without it I was getting a NoClassDefFoundError on org/hamcrest/MatcherAssert

A googling suggested that I needed a newer version of hamcrest.

**Comment by sectioneight** [comment]
Here is the error with an untouched `pom.xml`

```java
testRoundTripConfiguration(com.uber.jenkins.phabricator.PhabricatorBuildWrapperTest)  Time elapsed: 24.921 sec  <<< ERROR!
java.lang.NoClassDefFoundError: org/hamcrest/MatcherAssert
	at java.net.URLClassLoader.findClass(URLClassLoader.java:381)
	at java.lang.ClassLoader.loadClass(ClassLoader.java:424)
	at sun.misc.Launcher$AppClassLoader.loadClass(Launcher.java:331)
	at java.lang.ClassLoader.loadClass(ClassLoader.java:357)
	at org.junit.Assert.assertThat(Assert.java:865)
	at org.junit.Assert.assertThat(Assert.java:832)
	at org.jvnet.hudson.test.JenkinsRule$WebClient.goTo(JenkinsRule.java:2035)
	at org.jvnet.hudson.test.JenkinsRule$WebClient.goTo(JenkinsRule.java:2004)
	at org.jvnet.hudson.test.JenkinsRule$WebClient.getPage(JenkinsRule.java:1964)
	at com.uber.jenkins.phabricator.PhabricatorBuildWrapperTest.testRoundTripConfiguration(PhabricatorBuildWrapperTest.java:57)
```

**Comment by sectioneight** [comment]
Weird.. Travis is bailing:

```
Tests run: 9, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 3.746 sec - in InjectedTest
Results :
Tests run: 74, Failures: 0, Errors: 0, Skipped: 0
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time: 01:39 min
[INFO] Finished at: 2015-07-28T21:08:29+00:00
[INFO] Final Memory: 28M/452M
[INFO] ------------------------------------------------------------------------
[ERROR] Failed to execute goal org.apache.maven.plugins:maven-surefire-plugin:2.16:test (default-test) on project phabricator-plugin: ExecutionException; nested exception is java.util.concurrent.ExecutionException: java.lang.RuntimeException: The forked VM terminated without saying properly goodbye. VM crash or System.exit called ?
[ERROR] Command was/bin/sh -c cd /home/travis/build/uber/phabricator-jenkins-plugin && /usr/lib/jvm/java-7-openjdk-amd64/jre/bin/java -jar /home/travis/build/uber/phabricator-jenkins-plugin/target/surefire/surefirebooter1026712571483654866.jar /home/travis/build/uber/phabricator-jenkins-plugin/target/surefire/surefire2845929607535372867tmp /home/travis/build/uber/phabricator-jenkins-plugin/target/surefire/surefire_14673400948742534337tmp
```

**Pull request closed without merging**